cli/internal/cmdutil/cmdutil.go

```diff
@@ -0,0 +1,218 @@
+	}
+}
+
+func (h *Helper) getUI(flags *pflag.FlagSet) cli.Ui {
+	colorMode := ui.GetColorModeFromEnv()
+	if flags.Changed("no-color") && h.noColor {
+		colorMode = ui.ColorModeSuppressed
+	}
+	if flags.Changed("color") && h.forceColor {
+		colorMode = ui.ColorModeForced
+	}
+	return ui.BuildColoredUi(colorMode)
+}
+
+func (h *Helper) getLogger() (hclog.Logger, error) {
+	var level hclog.Level
+	switch h.verbosity {
+	case 0:
+		if v := os.Getenv(_envLogLevel); v != "" {
+			level = hclog.LevelFromString(v)
+			if level == hclog.NoLevel {
+				return nil, fmt.Errorf("%s value %q is not a valid log level", _envLogLevel, v)
+			}
+		} else {
+			level = hclog.NoLevel
+		}
+	case 1:
+		level = hclog.Info
+	case 2:
+		level = hclog.Debug
+	case 3:
+		level = hclog.Trace
+	default:
+		level = hclog.Trace
+	}
+	// Default output is nowhere unless we enable logging.
+	output := ioutil.Discard
+	color := hclog.ColorOff
+	if level != hclog.NoLevel {
+		output = os.Stderr
+		color = hclog.AutoColor
+	}
+
+	return hclog.New(&hclog.LoggerOptions{
+		Name:   "turbo",
+		Level:  level,
+		Color:  color,
+		Output: output,
+	}), nil
+}
+
+// AddFlags adds common flags for all turbo commands to the given flagset and binds
+// them to this instance of Helper
+func (h *Helper) AddFlags(flags *pflag.FlagSet) {
+	flags.BoolVar(&h.forceColor, "color", false, "Force color usage in the terminal")
+	flags.BoolVar(&h.noColor, "no-color", false, "Suppress color usage in the terminal")
+	flags.CountVarP(&h.verbosity, "verbosity", "v", "verbosity")
+	flags.StringVar(&h.rawRepoRoot, "cwd", "", "The directory in which to run turbo")
+	client.AddFlags(&h.clientOpts, flags)
+	config.AddRepoConfigFlags(flags)
+	config.AddUserConfigFlags(flags)
+}
+
+// NewHelper returns a new helper instance to hold configuration values for the root
+// turbo command.
+func NewHelper(turboVersion string) *Helper {
+	return &Helper{
+		TurboVersion:   turboVersion,
+		UserConfigPath: config.DefaultUserConfigPath(),
+	}
+}
+
+// GetCmdBase returns a CmdBase instance configured with values from this helper.
+// It additionally returns a mechanism to set an error, so
+func (h *Helper) GetCmdBase(flags *pflag.FlagSet) (*CmdBase, error) {
+	ui := h.getUI(flags)
+	logger, err := h.getLogger()
+	if err != nil {
+		return nil, err
+	}
+	cwd, err := fs.GetCwd()
+	if err != nil {
+		return nil, err
+	}
+	repoRoot := fs.ResolveUnknownPath(cwd, h.rawRepoRoot)
+	repoRoot, err = repoRoot.EvalSymlinks()
+	if err != nil {
+		return nil, err
+	}
+	repoConfig, err := config.ReadRepoConfigFile(config.GetRepoConfigPath(repoRoot), flags)
+	if err != nil {
+		return nil, err
+	}
+	userConfig, err := config.ReadUserConfigFile(h.UserConfigPath, flags)
+	if err != nil {
+		return nil, err
+	}
+	remoteConfig := repoConfig.GetRemoteConfig(userConfig.Token())
+	if remoteConfig.Token == "" && isCI() {
+		vercelArtifactsToken := os.Getenv("VERCEL_ARTIFACTS_TOKEN")
+		vercelArtifactsOwner := os.Getenv("VERCEL_ARTIFACTS_OWNER")
```

Comment:
@tknickman I'd love to think about how to make this configurable for any platform (e.g. if they implement their own remote cache).